### PR TITLE
Tests

### PR DESCRIPTION
**Pull request by fdansv**
Using Istambul's integration in Karma. `npm test` will create the html report in `./coverage/`

@rochoa

### CODE REVIEW

File: karma.conf.js [hunk 1]
```diff
@@ -32,13 +33,14 @@ module.exports = function(config) {
     // preprocess matching files before serving them to the browser
     // available preprocessors: https://npmjs.org/browse/keyword/karma-preprocessor
     preprocessors: {
+        'dist/d3.cartodb.js': ['coverage']
     },
 
 
     // test results reporter to use
     // possible values: 'dots', 'progress'
```
Comment (rochoa) [inline karma.conf.js:41]: b3qijlfccaejlwm

File: package.json [hunk 2]
```diff
@@ -30,13 +30,18 @@
   "homepage": "https://github.com/CartoDB/d3.cartodb",
   "dependencies": {
     "browserify": "^11.2.0",
-    "d3": "^3.5.6"
+    "d3": "^3.5.6",
+    "uglifyjs": "^2.4.10",
```
Comment (rochoa) [inline package.json]: This should be a dev dependency, right?

File: package.json [hunk 3]
```diff
@@ -30,13 +30,18 @@
...
+    "carto": "https://github.com/CartoDB/carto/archive/master.tar.gz",
```
Comment (rochoa) [inline package.json]: I would use a tag to have a reproducible build and a stable version.

File: src/renderer.js [hunk 4]
```diff
@@ -1,5 +1,7 @@
-var d3 = require('d3');
-var cartodb = cartodb || {};
+var d3 = window.d3 || require('d3');
```
Comment (rochoa) [inline src/renderer.js]: Not so important but all these should be `global` instead of `window`. See https://github.com/substack/browserify-handbook#global

**Comment by rochoa** [comment]
Small issues, none of them blocking. Feel free to merge.

**Comment by fdansv** [comment]
@rochoa I've enlarged the scope of this PR. Including unit tests and the setup of the test framework.

**Comment by fdansv** [comment]
@rochoa re: your comments. I'm addressing them now.

**Comment by fdansv** [comment]
@rochoa this is ready my side. Let's merge and move on to documenting?

**Comment by fdansv** [comment]
cc @javisantana too

**Comment by rochoa** [comment]
Can you update the branch so we can merge it?

**Comment by rochoa** [comment]
For next time: I think is better to open a new PR instead of increase scope.

File: spec/layer.spec.js [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+	// 		this.layer.renderer.render = oldRender;
+	// 		done()
+	// 	};
+	// 	this.layer.setCartoCSS("/** simple visualization */ #snow{ marker-fill-opacity: 0.9; marker-line-color: #FFF; marker-line-width: 1; marker-line-opacity: 1; marker-placement: point; marker-type: ellipse; marker-width: 10; marker-fill: #FF6600; marker-allow-overlap: true; }");
+	// })
+
+	it("'s tile size should always be a power of 2, equal or higher to 256", function(){
+		expect(this.layer._getTileSize()).not.toBeLessThan(256);
+		var log = Math.log2(this.layer._getTileSize())
+		expect(log).toEqual(parseInt(log, 10));
```
Comment (rochoa) [inline spec/layer.spec.js:74]: I don't understand this. What is this testing?

Comment (fdansv) [inline spec/layer.spec.js:74]: tests that the return value of gettilesize is a power of two integer higher or equal than 256

**Comment by fdansv** [comment]
Cool, all sorted, merging :)

File: spec/layer.spec.js (hunk 5 above)
Comment (rochoa) [inline spec/layer.spec.js:74]: What I meant is that `log` is gonna be always equal to `parseInt(log, 10)`, isn't it?

File: spec/layer.spec.js [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+		this.layer.addTo(this.map);
+		expect(this.layer._container.attributes["class"].value).toEqual('leaflet-tile-container leaflet-zoom-animated');
+		expect(this.layer._container.parentNode.attributes["class"].value).toEqual('leaflet-layer');
+	});
+
+	it("should run onRemove when layer is removed from map", function() {
+		this.layer.addTo(this.map);
+		spyOn(this.layer, "onRemove");
+		this.map.removeLayer(this.layer);
+		expect(this.layer.onRemove).toHaveBeenCalled;
```
Comment (rochoa) [inline spec/layer.spec.js:57]: Isn't his missing the function call in `toHaveBeenCalled`?

Comment (fdansv) [inline spec/layer.spec.js:57]: Yes it is.

File: spec/layer.spec.js (hunk 5 above)
Comment (fdansv) [inline spec/layer.spec.js:74]: If the tile size comes as, say, 280, its `log2` will not be an integer, so it will fail the test.

Comment (rochoa) [inline spec/layer.spec.js:74]: True story, I totally miss the `parseInt` thing :-( Sorry about the noise.

Comment (fdansv) [inline spec/layer.spec.js:74]: :wink:

**Pull request merged**